### ISSUE TITLE
Allocation failure: trying to chainload Linux on aarm64

### ISSUE THREAD

**Issue by dimaqq** [Limine-Bootloader/Limine#302]
I had a working environment with EFI_STUB Linux 6.4.0+, root UUID and initrd.

I've tried to upgrade it to Limine to allow me to switch kernels easier...

However, when I get the syntax right (I think), I end up with:

```
PANIC: chainload: Allocation failure
Stacktrace:
  [0xbc12ceac] <panic+0xb4>
  [0xbc129ebc] <efi_chainload_file+0x230>
  [0xbc129c8c] <chainload+0x6c>
  [0xbc126474] <boot+0x1d8>
  [0xbc125ec8] <_menu+0xd64>
End of trace. Press a key to return to menu.
```

Environment:

UTM on Mac

```
>>> version
5.20230909.0
```

Manually edited entry that triggers this error:

```
?PROTOCOL=chainload                                                                ?
?IMAGE_PATH=boot:///Image                                                     ?
?CMDLINE=root=UUID=UUID=4c3e28ab-713f-4718-8a9b-434440560ce9 rw                ?
?MODULE_PATH=boot:///initramfs-linux.img  
```

The files exist, `Image` is only `42889728` in size.

**Comment by Mintsuki** [comment]
Did this also happen on `5.20230830.0`?

**Comment by dimaqq** [comment]
Sorry I'm not sure I have a good way to test.
At the moment the VM cannot be booted, so I'll have to recreated the env.
I can try tomorrow maybe.

**Comment by qookei** [comment]
How much RAM did you assign to the VM?

**Comment by qookei** [comment]
Also worth noting, `MODULE_PATH` doesn't work with EFI chainloading, you need to specify the path to the initramfs on the kernel command line.

**Comment by Mintsuki** [comment]
The issue should be fixed as of 277cec6c8876013a18851703e07e8cfc9c1298fb, please reopen if that's not the case.

**Comment by dimaqq** [comment]
Thank you! :)
I'll retest a bit later when arch linux catches up or maybe I'll build from source.
